Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM golang:1.20.2 as builder
+FROM --platform=$TARGETPLATFORM docker.io/library/golang:1.20.4 as builder
```

Comment:
is this not picked up automatically?

Reply:
No idea, this is what we have in https://github.com/elastic/cloud-on-k8s/blob/main/Dockerfile#L2

Actually after having a look at the [documentation](https://docs.docker.com/build/building/multi-platform/) I'm wondering if it was a good idea to remove the `ARGS` in https://github.com/elastic/elasticsearch-k8s-metrics-adapter/pull/28/commits/c300801d01db371110791c0821a56e6e42132f61:

> Finally, depending on your project, the language that you use may have good support for cross-compilation. In that case, multi-stage builds in Dockerfiles can be effectively used to build binaries for the platform specified with --platform using the native architecture of the build node.

@thbkrkr sorry for the ping, any idea if it is required here?

Reply:
sorry for the noise,
there are surely several ways to do docker cross-compilation.
Just follow one that you find the most suiteable

Reply:
ah right, if your compiler like go supports it, you could do just this:

`FROM --platform=$BUILDPLATFORM golang:alpine AS build`

This will pull the same layers as your local machine (`BUILDPLATFORM`). Then you have to pass the  TARGETPLATFORM ( as specified in `buildx --paltform` to the compiler.
`RUN compile --target=$TARGETPLATFORM -o /out/mybinary`

But you can leave this out. Then it will pick even for the builder the `TARGETPLATFORM`.
However, it might need to pull multiple images

Reply:
I'm not 100% sure but I believe this is necessary when building with dockerx an image for a platform different from the host platform. This should not be necessary for native builds.

Reply:
Thanks both, I understand now. Since I already spent a fair amount of time on this one, and unless there is a strong concern about the proposed way to compile the adapter (i.e. using `--platform=$TARGETPLATFORM`), I think I would keep it as is. We can improve things later by using the native cross compilation feature from Go.